triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/tsprovisioning/transformation/namedgraphs/projects/UpdatesCreator.scala

```diff
@@ -170,4 +171,33 @@ private object UpdatesCreator extends UpdatesCreator {
         )
       }
       .toList
+
+  override def postUpdates(project: Project): List[SparqlQuery] =
+    List(deduplicateDateCreated(project))
+
+  /** Retain the lowest dateCreated date. */
+  private def deduplicateDateCreated(project: Project) = {
+    val resource = project.resourceId.showAs[RdfResource]
+    SparqlQuery.of(
+      name = "transformation - project date created deduplicate",
+      Prefixes.of(schema -> "schema"),
+      s"""|WITH $resource 
+          |DELETE { ?p schema:dateCreated ?date }
+          |WHERE {
+          |  {
+          |    SELECT (min(?cdate) as ?minDate)
+          |    WHERE {
+          |      ?id a schema:Project;
```

Comment:
It's a minor thing but perhaps selecting the `?id` in the subquery could allow you to use it in line 196 so you don't even have to check the type but select the date.